src/stream/src/executor/project_set.rs

```diff
@@ -174,22 +184,33 @@ impl Inner {
                                 unsafe { std::mem::transmute(row.as_mut_slice()) };
 
                             row[0] = Some(projected_row_id.into());
-                            // if any of the set columns has a value
-                            let mut valid = false;
+
+                            // Whether all table functions has exhausted or has failed for current input row.
+                            let mut fully_consumed = true;
```

Comment:
In which case it's considered as "exhausted"?

Reply:
When this condition doesn't meet:

```rust
if let Some((i, result)) = state.peek()
  && is_current_input(i)
```

i.e., the cursor of the `state` has moved to the next input.